bakery/checkers/checkers_test.go

```diff
@@ -418,3 +418,49 @@ func (*CheckersSuite) TestInferDeclared(c *gc.C) {
 		c.Assert(checkers.InferDeclared(ms), jc.DeepEquals, test.expect)
 	}
 }
+
+var operationCheckerTests = []struct{
+	about string
+	caveat checkers.Caveat
+	operation string
+	expectError string
+}{{
```

Comment:
Let's have some tests with zero numbers of operations too, please.
In fact, perhaps AllowCaveat with no operations could return an 
ErrorCaveat saying "no operations allowed".